src/Swift.Bindings/src/Model/TypeNames/CSharpTypeName.cs

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace BindingsGeneration;
+
+/// <summary>
+/// Represents a C# type name.
+/// </summary>
+public record CSharpTypeName
+{
+    /// <summary>
+    /// The namespace name.
+    /// </summary>
+    public string Namespace { get; }
+
+    /// <summary>
+    /// The type name.
+    /// </summary>
+    public string Name { get; }
+
+    /// <summary>
+    /// The fully qualified type name, including namespace.
+    /// </summary>
+    public string FullyQualifiedName { get; }
+
+    /// <inheritdoc />
+    public override string ToString() => FullyQualifiedName;
+
+    private CSharpTypeName(string @namespace, string name, string fullyQualifiedName)
+    {
+        Namespace = @namespace;
+        Name = name;
+        FullyQualifiedName = fullyQualifiedName;
+    }
+
+    /// <summary>
+    /// Creates a new CSharpTypeName from a namespace and type name.
+    /// </summary>
+    /// <param name="namespace">The namespace.</param>
+    /// <param name="name">The type name.</param>
+    /// <returns>The CSharpTypeName.</returns>
+    public static CSharpTypeName FromNamespaceAndName(string @namespace, string name)
+    {
+        ArgumentException.ThrowIfNullOrEmpty(name, nameof(name));
+        ArgumentException.ThrowIfNullOrEmpty(@namespace, nameof(@namespace));
+
+        if (name.Contains('<'))
+        {
+            throw new ArgumentException("Cannot create a CSharpTypeName from a generic type.");
+        }
+
+        var fullyQualifiedName = $"{@namespace}.{name}";
+        return new CSharpTypeName(@namespace, name, fullyQualifiedName);
+    }
+
+    /// <summary>
+    /// C# type name for void.
+    /// </summary>
+    public static readonly CSharpTypeName VoidType = new CSharpTypeName("", "", "void");
```

Comment:
Shouldn't this be `new CSharpTypeName("System", "Void", "System.Void");` ?
I guess it depends if we want to model this as a semantic model versus syntax.

Similarly - what will string look like? Or int, ...
Definitely worth a comment.